xbmc/addons/kodi-dev-kit/include/kodi/tools/Timer.h

```diff
@@ -0,0 +1,315 @@
+/*
+ *  Copyright (C) 2005-2020 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#ifdef __cplusplus
+
+#include "Thread.h"
+
+#include <functional>
+
+namespace kodi
+{
+namespace tools
+{
+
+//==============================================================================
+/// @defgroup cpp_kodi_tools_CTimer class CTimer
+/// @ingroup cpp_kodi_tools
+/// @brief **Time interval management**\n
+/// Class which enables a time interval to be called up by a given function or
+/// class by means of a thread.
+///
+/// His code uses the support of platform-independent thread system introduced
+/// with C++11.
+///
+///
+/// ----------------------------------------------------------------------------
+///
+/// **Example:**
+/// ~~~~~~~~~~~~~{.cpp}
+/// #include <kodi/tools/Timer.h>
+///
+/// class ATTRIBUTE_HIDDEN CExample
+/// {
+/// public:
+///   CExample() : m_timer([this](){TimerCall();})
+///   {
+///     m_timer.Start(5000, true); // let call continuously all 5 seconds
+///   }
+///
+///   void TimerCall()
+///   {
+///     fprintf(stderr, "Hello World\n");
+///   }
+///
+/// private:
+///   kodi::tools::CTimer m_timer;
+/// };
+/// ~~~~~~~~~~~~~
+///
+///@{
+class CTimer : protected CThread
```

Comment:
Just `Timer`, not `CTimer`?

Reply:
Just `Timer` will cause conflicts with some PVR add-one I would say.